### PR TITLE
Feature/passport visa

### PR DESCRIPTION
**Pull request by Azher2Ali**
This PR contains changes related to the below items :

1. Create a visa service that will : #703 

- parseVisa → Visa class that is returned from content of Visa JWT (string)
- validateVisa → need public key? or do we just trust the passport broker (i think we just trust the broker so these are already valid)
- Caching pubic key using Spring Cache (CacheUtil) fetched from passport broker getPublicKey endpoint (maintained at application.yaml) and evicting every midnight
- getPermissionsForVisa → return list of permissions for visa

2. Create a Passport Service : #701 

- parsePassport → return Passport object from content of Passport JWT (string)
- validatePassport (vs public key) → requires passportBroker config (to fetch public key) 
- Caching pubic key using Spring Cache (CacheUtil) fetched from passport broker getPublicKey endpoint (maintained at application.yaml) and evicting every midnight
- getVisas → return list of visa JWTs
- getPermissionsForPassport → return list of permissions from passport (compiled list of all permissions from all visas, remove duplicates)

image

### CODE REVIEW

File: src/main/java/bio/overture/ego/service/PassportService.java [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+              if (visaService.getById(visa.getId()) != null) {
+                visaPermissions.addAll(visaPermissionService.getPermissionsForVisa(visa));
+              }
+            });
+    return visaPermissions;
+  }
+
+  // Parse Passport token to extract the passport body
+  public Passport parsePassport(@NonNull String passportJwtToken) throws JsonProcessingException {
+    String[] split_string = passportJwtToken.split("//.");
```
Comment (leoraba) [inline src/main/java/bio/overture/ego/service/PassportService.java]: regex syntax uses backlash to scape special characters. in this case should be like "\\."

File: src/main/java/bio/overture/ego/service/VisaService.java [hunk 2]
```diff
@@ -65,6 +66,18 @@ public void delete(@NonNull UUID id) {
     super.delete(id);
   }
 
+  // Parses Visa JWT token to convert into Visa Object
+  public Visa parseVisa(@NonNull String visaJwtToken) throws JsonProcessingException {
+    String[] split_string = visaJwtToken.split("//.");
```
Comment (leoraba) [inline src/main/java/bio/overture/ego/service/VisaService.java]: regex syntax uses backlash to scape special characters. in this case should be like `\\.`

File: src/main/java/bio/overture/ego/model/dto/Passport.java [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+import lombok.AllArgsConstructor;
+import lombok.Builder;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+
+@Data
+@Builder
+@AllArgsConstructor
+@NoArgsConstructor
+public class Passport {
```
Comment (leoraba) [inline src/main/java/bio/overture/ego/model/dto/Passport.java:16]: Passport could contain additional/optional fields as per [spec](https://ga4gh.github.io/data-security/aai-openid-connect-profile#passport-format)

consider adding mapping annotation to ignore unknown fields:
`@JsonIgnoreProperties(ignoreUnknown = true)`

File: src/main/java/bio/overture/ego/service/PassportService.java (hunk 1 above)
Comment (Azher2Ali) [reply src/main/java/bio/overture/ego/service/PassportService.java]: Resolved

File: src/main/java/bio/overture/ego/model/dto/Passport.java (hunk 3 above)
Comment (Azher2Ali) [reply src/main/java/bio/overture/ego/model/dto/Passport.java:16]: Added JsonIgnoreProperties

File: src/main/resources/application.yml [hunk 4]
```diff
@@ -184,6 +184,12 @@ token:
...
...FKT0bW/svFUTjXgGnZy1ncz9GpENpMH3lQDQVibteKpYwcom+Cr0XlQ66VUcudPrDjcOY7vhuMfnSh1YWLYyM4IeRHtcUxDVkFoM+vEFNHLf2zIOqqbgmboW3iDVIurT7iRO7KxAe/YtWJL9aVqMtBn7Lu7S7OvAU4ji5iLIBxjl82JYA+9lu/aQ6YGaoZuSO7bcU8Sivi+DKAahqN9XMKiB1XpC+PpaS/aec2S7xIlTdzoDGxEALRGlMe+xBEeQTBVJHBWrRIDPoHLTREeRC/9Pp+1Y4Dz8hd5Bi0n8/5r/q0liD+0vtmjsdU4E2QrktYECgYEA73qWvhCYHPMREAFtwz1mpp9ZhDCW6SF+njG7fBKcjz8OLcy15LXiTGc268ewtQqTMjPQlm1n2C6hGccGAIlMibQJo3KZHlTs125FUzDpTVgdlei6vU7M+gmfRSZed00J6jC04/qMR1tnV3HME3np7eRTKTA6Ts+zBwEvkbCetSkCgYEA4NY5iSBO1ybouIecDdD15uI2ItLPCBNMzu7IiK7IygIzuf+SyKyjhtFSR4vEi0gScOM7UMlwCMOVU10e4nMDknIWCDG9iFvmIEkGHGxgRrN5hX1Wrq74wF212lvvagH1IVWSHa8cVpMe+UwKu5Q1h4yzuYt6Q9wPQ7Qtn5emBE0CgYB2syispMUA9GnsqQii0Xhj9nAEWaEzhOqhtrzbTs5TIkoA4Yr3BkBY5oAOdjhcRBWZuJ0XMrtaKCKqCEAtW+CYEKkGXvMOWcHbNkkeZwv8zkQ73dNRqhFnjgVn3RDNyV20uteueK23YNLkQP+KV89fnuCpdcIw9joiqq/NYuIHoQKBgB5WaZ8KH/lCA8babYEjv/pubZWXUl4plISbja17wBYZ4/bl+F1hhhMr7Wk//743dF2NG7TT6W0VTvHXr9IoaMP65uQmKgfbNpsGn294ZClGEFClz+t0KpZyTpZvL0fjibr8u+GLfkxkP5qt2wjif7KRlrKjklTTva+KAVn2cW1FAoGBAMkX9ekIwhx/7uY6ndxKl8ZMDerjr6MhV0b08hHp3RxHbYVbcpN0UKspoYvZVgHwP18xlDij8yWRE2fapwgi4m82ZmYlg0qqJmyqIU9vBB3Jow903h1KPQrkmQEZxJ/4H8yrbgVf2HT+WUfjTFgaDZRl01bI3YkydCw91/Ub9HU6
```
Comment (leoraba) [inline src/main/resources/application.yml]: we shouldn't keep our private/public keys on git repo.

Comment (Azher2Ali) [reply src/main/resources/application.yml]: The pubic/private key used here was a dummy one which was required during testing. As per the requested change, following keys are being removed from the git repo

**Review by leoraba — APPROVED** [review]
Looks good to me. 👍

**Review by leoraba — APPROVED** [review]
👍 approved!

**Pull request merged**